packages/h5wasm/src/h5wasm-api.ts

```diff
@@ -135,21 +146,59 @@ export class H5WasmApi extends DataProviderApi {
     return [];
   }
 
-  private async initFile(buffer: ArrayBuffer): Promise<H5WasmFile> {
-    const { FS } = await h5wasmReady;
+  private async initH5Wasm(): Promise<typeof Module> {
+    const module = await h5wasmReady;
+
+    // Replace default plugins path
+    module.remove_plugin_search_path(0);
+    module.insert_plugin_search_path(PLUGINS_PATH, 0);
 
-    // `FS` is guaranteed to be defined once H5Wasm is ready
-    // https://github.com/silx-kit/h5web/pull/1082#discussion_r858613242
-    assertNonNull(FS);
+    // Create plugins folder on Emscripten virtual file system
+    // @ts-expect-error
+    module.FS.mkdirTree(PLUGINS_PATH);
 
-    // Write file to Emscripten virtual file system
+    return module;
+  }
+
+  private async initFile(buffer: ArrayBuffer): Promise<H5WasmFile> {
+    const h5Module = await this.h5wasm;
+
+    // Write HDF5 file to Emscripten virtual file system
     // https://emscripten.org/docs/api_reference/Filesystem-API.html#FS.writeFile
     const id = nanoid(); // use unique ID instead of `this.filepath` to avoid slashes and other unsupported characters
-    FS.writeFile(id, new Uint8Array(buffer), { flags: 'w+' });
+    h5Module.FS.writeFile(id, new Uint8Array(buffer), { flags: 'w+' });
 
     return new H5WasmFile(id, 'r');
   }
 
+  private async processFilters(filters: Filter[]): Promise<void> {
+    const h5Module = await this.h5wasm;
+
+    for await (const filter of filters) {
+      if (filter.id < h5Module.H5Z_FILTER_RESERVED) {
+        continue; // filter supported out of the box
+      }
+
+      const plugin = PLUGINS_BY_FILTER_ID[filter.id];
+      if (!plugin) {
+        throw new Error(
+          `Compression filter ${filter.id} not supported (${filter.name})`,
+        );
+      }
+
+      const pluginPath = `${PLUGINS_PATH}/libH5Z${plugin}.so`;
+
+      if (h5Module.FS.analyzePath(pluginPath).exists) {
+        continue; // plugin already loaded
+      }
+
+      const buffer = await this.getPlugin?.(plugin);
+      if (buffer) {
```

Comment:
Shouldn't you return an error if `buffer` is undefined ? 

Even if the plugin exists in `PLUGINS_BY_FILTER_ID`, it may be that the `getPlugin` method does not implement it and you may want to notify the user about it.

Reply:
True! I didn't want to throw when `getPlugin` is not implemented, and this threw me off :drum:

Reply:
Done. The error message is slightly different from the other one to make sure we can distinguish them.